src/backend/distributed/commands/vacuum.c

```diff
@@ -541,6 +564,14 @@ VacuumStmtParams(VacuumStmt *vacstmt)
 		#endif
 		else if (strcmp(opt->defname, "index_cleanup") == 0)
 		{
+			#if PG_VERSION_NUM >= PG_VERSION_14
+			if (opt->arg && strcmp(defGetString(opt), "auto") == 0)
```

Comment:
this part seems a little hard to follow. Can't we decide `params.index_cleanup` in a single `if` or `case` statement? Now that the logic seems scattered

Reply:
First, we should follow the same logic as PG: https://github.com/postgres/postgres/blob/07d683b54af854098cc559d4b8640905f9efa0ea/src/backend/commands/vacuum.c#L151-L166


Such that `if (!opt->arg)` is also interpreted as `auto` or `auto` should be checked case insensitive via `pg_strcasecmp`. Can you also add a tests for these case? 

In terms of code, can't we do the following where we make this difference a lot more explicit to the user:
```C
    
		else if (strcmp(opt->defname, "index_cleanup") == 0)
		{
#if PG_VERSION_NUM >= PG_VERSION_14

			/* Interpret no string as the default, which is 'auto' */
			if (!opt->arg)
				params.index_cleanup = VACOPTVALUE_AUTO;
			else
			{
				char	   *sval = defGetString(opt);

				/* Try matching on 'auto' string, or fall back on boolean */
				if (pg_strcasecmp(sval, "auto") == 0)
					params.index_cleanup = VACOPTVALUE_AUTO;
				else
					params.index_cleanup = get_vacoptval_from_boolean(opt);
			}
		}
#else    
... /* old code */
#endif  
```

We can omit `get_vacoptval_from_boolean` function for this PR, but would be good clean-up for the next ones